File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    rank = {'R':0, 'T':0, 'C':0, 'F':0, 'J':0, 'M':0, 'A':0, 'N':0}
    for i in range(len(survey)):
        if choices[i]-1 > 3:
            rank[survey[i][1]] += (choices[i]-4)
        elif choices[i]-1 < 3:
            if choices[i] == 1:
                rank[survey[i][0]] += 3
            elif choices[i] == 2:
                rank[survey[i][0]] += 2
            elif choices[i] == 3:
                rank[survey[i][0]] += 1
    if rank['R'] >= rank['T']:
        answer+='R'
    else:
        answer+='T'
    if rank['C'] >= rank['F']:
        answer+='C'
    else:
        answer+='F'
    if rank['J'] >= rank['M']:
        answer+='J'
    else:
        answer+='M'
    if rank['A'] >= rank['N']:
        answer+='A'
    else:
        answer+='N'
    return answer

File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    let answer = ''
    
    const test = {
        "R" : 0,
        "T" : 0,
        "F" : 0,
        "C" : 0,
        "M" : 0,
        "J" : 0,
        "A" : 0,
        "N" : 0,
    }
    
    const choose = (val, num) => {
        const val1 = val[0], val2 = val[1]
        switch(num) {
            case 1:
                return test[val1]+=3;
            case 2:
                return test[val1]+=2;
            case 3:
                return test[val1]+=1;
            case 4:
                return test[val1]+=0;
            case 5:
                return test[val2]+=1;
            case 6:
                return test[val2]+=2;
            case 7:
                return test[val2]+=3;
        }
    }

    for(let i=0; i < survey.length; i++){
        choose(survey[i], choices[i])
    }
    
    if(test['R'] >= test['T']) answer+='R';
    else answer+='T';
    
    if(test['C'] >= test['F']) answer+='C';
    else answer+='F';
    
    if(test['J'] >= test['M']) answer+='J';
    else answer+='M';
    
    if(test['A'] >= test['N']) answer+='A';
    else answer+='N';
    
    return answer
}